plugins/BEdita/API/tests/TestCase/Auth/EndpointAuthorizeTest.php

```diff
@@ -134,20 +136,40 @@ public function getEndpointProvider()
                 new Uri('/home/sweet/home'),
             ],
             '/' => [
-                null,
+                new Endpoint(
+                    [
+                        'name' => '',
+                        'enabled' => true
+                    ],
+                    [
+                        'source' => 'Endpoints'
+                    ]
+                ),
                 new Uri('/'),
             ],
             '/this/endpoint/definitely/doesnt/exist' => [
-                null,
+                new Endpoint(
+                    [
+                        'name' => 'this',
+                        'enabled' => true
+                    ],
+                    [
+                        'source' => 'Endpoints'
+                    ]
+                ),
                 new Uri('/this/endpoint/definitely/doesnt/exist'),
             ],
+            '/disabled/endpoint' => [
+                new NotFoundException('Resource not found.'),
+                new Uri('/disabled/endpoint'),
+            ]
         ];
     }
 
     /**
      * Test getting endpoint from request.
      *
-     * @param int|\Exception $expected Expected endpoint ID.
+     * @param int|\Cake\ORM\Entity $expected Expected endpoint ID or entity.
```

Comment:
To be (_very_) pedantic, this parameter's type is `int|\Cake\ORM\Entity|\Exception`. Or `mixed` to be more concise. 🙃